starlingcaptureapi/asset_helper.py

```diff
@@ -167,6 +171,31 @@ def get_internal_claim_fullpath(self, from_file):
             self.dir_internal_claims, _file_util.digest_sha256(from_file) + ".json"
         )
 
+    def _shared_collection_prefix(self, collection_id: str) -> str:
+        return os.path.join(self.shared_prefix, collection_id)
+
+    def legacy_path_for(self, action: str, output: bool = False) -> str:
+        """Returns a full directory path for the given action."""
+        return os.path.join(
+            self.shared_prefix,
+            f"{action}-output" if output else action,
+        )
+
+    def path_for(self, collection_id: str, action: str, output: bool = False):
+        """Retuns a full directory path for the given collection and action.
+
+        Appends `-output` if output=True.
+        """
+        # Backwards-compatibility workaround for missing collection_ids.
+        # Can be removed once legacy paths are no longer in use.
+        if collection_id is None:
+            return self.legacy_path_for(action, output=output)
+
+        return os.path.join(
+            self._shared_collection_prefix(collection_id),
+            f"{action}-output" if output else action,
+        )
```

Comment:
In [this comment](https://github.com/starlinglab/starling-integrity-api/issues/67#issuecomment-1071479089) about the archive action, @benhylau mentioned paths like `collection_id/action-archive/`. Is that out of date, and now the correct path would be `collection_id/archive-output/`? Or should this code be changed?

Reply:
This code merely preserves the existing behavior and naming scheme for actions. For the already-existing actions, there are two directories for each action, an "input" one and an "output" one. The output one has currently `-output` appended to it.

If/when y'all decided to change that scheme, you can update this code (and migrate any associated data).